### ISSUE TITLE
registry-console dc is missing default nodeSelector

### ISSUE THREAD

**Issue by etsauer** [openshift/openshift-ansible#2879]
When running a fairly vanilla, 1 master, several nodes install, I noticed that the `registry-console` pod deploys to one of the "app" nodes. I would expect that it should deploy to a node labelled with `region=infra` as my docker-registry pod does. Upon looking at the deployment config, my `dc/registry-console` does not have a `nodeSelector` configured. More info below.

##### Version
```
$ git describe
openshift-ansible-3.4.17-1-95-ga8b36be
$ ansible --version
ansible 2.1.1.0
  config file = /home/esauer/.ansible.cfg
  configured module search path = Default w/o overrides
```
##### Steps To Reproduce
run `ansible-playbook -i hosts /path/to/openshift-ansible/playbooks/byo/config.yaml` with a simple inventory like:

```
# Create an OSEv3 group that contains the master, nodes, etcd, and lb groups.
# The lb group lets Ansible configure HAProxy as the load balancing solution.
# Comment lb out if your load balancer is pre-configured.
[OSEv3:children]
masters
nodes
etcd

# Set variables common for all OSEv3 hosts
[OSEv3:vars]

deployment_type=openshift-enterprise
openshift_deployment_type=openshift-enterprise

openshift_master_default_subdomain=apps.test-stack.casl.rht-labs.com

# Uncomment the following to enable htpasswd authentication; defaults to
# DenyAllPasswordIdentityProvider.
openshift_master_identity_providers=[{'name': 'htpasswd_auth', 'login': 'true', 'challenge': 'true', 'kind': 'HTPasswdPasswordIdentityProvider', 'filename': '/etc/origin/master/htpasswd'}]


# Host Groups
[masters]  
master-0.d1.casl.rht-labs.com

[etcd:children]
master-0.d1.casl.rht-labs.com

[infra_nodes]
infranode-[0:1].d1.casl.rht-labs.com
[infra_nodes:vars]
openshift_node_labels="{'region':'infra','zone':'default'}"

[app_nodes]
app-node-[0:3].d1.casl.rht-labs.com
[app_nodes:vars]
openshift_node_labels="{'region':'app','zone':'default'}"

[nodes:children]
masters
infra_nodes
app_nodes
```

##### Current Result

```
[openshift@master-0 ~]$ oc get pods -o wide
NAME                       READY     STATUS    RESTARTS   AGE       IP              NODE
docker-registry-2-6lkde    1/1       Running   0          18m       10.129.0.3      infranode-0.d1.casl.rht-labs.com
registry-console-1-9foj4   1/1       Running   0          18m       10.130.2.3      app-node-3.d1.casl.rht-labs.com <---- registry console should show on infranode
router-1-3tgcu             1/1       Running   0          18m       192.168.99.8    infranode-1.d1.casl.rht-labs.com
router-1-fepa1             1/1       Running   0          18m       192.168.99.10   infranode-0.d1.casl.rht-labs.com

[openshift@master-0 ~]$ oc get dc docker-registry -o yaml | grep -A 1 nodeSelector
      nodeSelector:
        region: infra
[openshift@master-0 ~]$ oc get dc registry-console -o yaml | grep -A 1 nodeSelector
[openshift@master-0 ~]$  ### missing nodeSelector field ####

```

##### Expected Result
Output looking more like:

```
[openshift@master-0 ~]$ oc get pods -o wide
NAME                       READY     STATUS    RESTARTS   AGE       IP              NODE
docker-registry-2-6lkde    1/1       Running   0          18m       10.129.0.3      infranode-0.d1.casl.rht-labs.com
registry-console-1-9foj4   1/1       Running   0          18m       10.130.2.3      infranode-1.d1.casl.rht-labs.com <---- registry console should show on infranode
router-1-3tgcu             1/1       Running   0          18m       192.168.99.8    infranode-1.d1.casl.rht-labs.com
router-1-fepa1             1/1       Running   0          18m       192.168.99.10   infranode-0.d1.casl.rht-labs.com

[openshift@master-0 ~]$ oc get dc docker-registry -o yaml | grep -A 1 nodeSelector
      nodeSelector:
        region: infra
[openshift@master-0 ~]$ oc get dc registry-console -o yaml | grep -A 1 nodeSelector
      nodeSelector:
        region: infra
[openshift@master-0 ~]$
```

##### Additional Information
```
[openshift@master-0 ~]$ openshift version
openshift v3.3.1.5
kubernetes v1.3.0+52492b4
etcd 2.3.0+git
```

**Comment by garethahealy** [comment]
IHAC also hitting this issue. Can be fixed by patching the DC post install, but would rather not add work arounds.

**Comment by sdodson** [comment]
What's the concern with deploying the registry console anywhere? It should be a fairly light weight web front end pod. The registry itself on the other hand should definitely be deployed strategically.

**Comment by etsauer** [comment]
The concern is simply that most people will expect the console to follow
the registry, and would see it as a bug if it lands on a node that's
designated for applications. There's also the possibility that I may build
nodes differently if they will run production workloads vs, infrastructure
components that I'm OK taking outages on and I wouldn't want those to
workloads ty be allowed to mix.

**Comment by sdodson** [comment]
Thanks, I'll see if we can get this fixed for 3.5 release and then backport if feasible.

**Comment by 7rack** [comment]
does release-3.6 branch fix it?
